### PR TITLE
Added SANs support

### PR DESCRIPTION
**Pull request by SadriG91**
# Pull Request

## Description

Added SANs support and included an example using it.

Fixes # (issue)

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

### Checklist

- [x] `terraform fmt` and `terraform validate` both work from the root and `examples/` directories (look in CI for an example)
- [x] Provide an updated example that utilizes newly created resources, or for more advanced additions a new example under the examples directory.
- [ ] Example plan output in this PR (in lieu of CI)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Any breaking changes are noted in the description above
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] My code follows the style guidelines of this project
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -11,6 +11,12 @@ variable "certificate_name" {
   type        = string
 }
 
+variable "subject_alternative_names" {
+  description = "(Optional) Set of domains that should be SANs in the issued certificate."
+  type        = list(any)
```
Comment (larstobi) [inline variables.tf]: This should be:
`list(string)`

File: Taskfile.yml [hunk 2]
```diff
@@ -24,24 +24,25 @@ tasks:
...
       - |
+        set -e
         BOLD=$(tput bold)
         NORM=$(tput sgr0)
         CWD=$PWD
         for d in $DIRECTORIES; do
           cd $d
           echo "${BOLD}$PWD:${NORM}"
           if ! terraform fmt -check=true -list=false -recursive=true; then
-            echo "  ✗ terraform fmt" && exit $?
```
Comment (larstobi) [inline Taskfile.yml]: I think `$?` is more correct.

**Review by larstobi — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (SadriG91) [reply variables.tf]: done

File: Taskfile.yml (hunk 2 above)
Comment (SadriG91) [reply Taskfile.yml]: done

**Review by larstobi — APPROVED** [review]

**Pull request merged**